src/intersections/quadrangle.jl

```diff
@@ -0,0 +1,33 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    intersecttype(f, segment_or_ray, quadrangle)
+
+Calculate the intersection type of `Segment` or `Ray` and `Quadrangle` and apply function
+`f` to it.
+"""
+function intersecttype(
+  f::Function, segment_or_ray::Union{Segment{3,T},Ray{3,T}}, n::Quadrangle{3,T}
```

Comment:
I think we need to organize this better. We don't want to write it in terms of a Union type. Also, I think the internal implementation should be expressed in terms of intersecttype so that the function `f` can be forwarded. More generally, I think we should consider and `Ngon` as input, triangulate this ngon with the existing triangulation methods, and check the intersection.